docs/decisions/0005-event-schema-serialization-and-evolution.rst

```diff
@@ -0,0 +1,75 @@
+
+However, for the event bus, we will be introducing more rigorous event evolution rules using an explicit schema and schema registry. We have decided to use Avro elsewhere (`OEP-52/decisions/001-schema-representation.rst <https://github.com/openedx/open-edx-proposals/pull/233/files#diff-70c71499189a23f546da507c5bfdc0fea674f4cbbc5c039298d8390d6930a5ca>`_).
+
+Event schemas can evolve in various ways and each implementation of an event bus should choose a schema evolution configuration. The choices include how data can change, as well as the order in which producer and consumer needs to be deployed with new changes.
+
+Event bus technologies support various schema evolutions::
+
+- Backward / Backward Transitive: Allows you to delete fields and add optional fields. The consumer of the messages must upgrade to handle new schema before producer.
+
+- Forward / Forward Transitive: Allows you to add fields and delete optional fields. The producer of the messages must upgrade to handle new schema before consumer of  messages.
+
+- Full / Full Transitive: Allows you to add or delete optional fields. Either producer or consumer can change first.
+
+
+Currently, the schema for event bus messages in defined through two classes: OpenEdxPublicSignal and attrs decorated classes in data.py modules (`example <https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/data.py>`_).
+
+OpenEdxPublicSignal defines all the information necessary to create, serialize, and send an event. This includes the data to be sent and the metadata. The data to be sent is defined as the 'init_data' attribute. 'init_data' is a dict of key/value pairs. The keys are strings and the values can be of various types (the exact supported types has not be specified yet). These keys correspond to keyword arguments sent via the signal.
+
+The attrs decorated classes are used to structure the data sent. Hopefully, most of the values in 'init_data' attribute in OpenEdxPublicSignal are of these attrs decorated classes.
+Some helpful information about OpenEdxPublicSignal class:
+
+
+Decision
+--------
+
+- The ability to add new data to an event definition is being prioritized over the ability to delete information. Deleting will not be allowed without introducing breaking changes.
+
+- The top-level data dict of an OpenEdxPublicSignal can be evolved by adding new key/values. At this time, they will all be considered required.
+
+- An attr Data class will not be evolved once in use. If new fields are required for an event, they can only be added into the top-level data dict.
+
+- The above decisions result in a decision to use “Forward / Forward Transitive” schema evolution for external events. Although, it is possible to use “Full / Full Transitive” schema evolution and defer adjusting the rule until schema changes are required, or we add more flexibility.
```

Comment:
Removing the second part (which I had written) simply because it is confusing, and probably not worth the effort of clarifying.
```suggestion
- The above decisions result in a decision to use “Forward / Forward Transitive” schema evolution for external events.
```